ql/src/test/org/apache/hadoop/hive/ql/exec/vector/TestVectorOperationProcess.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.hadoop.hive.ql.exec.vector.expressions.gen.FilterLongColGreaterLongColumn;
+import org.apache.hadoop.hive.ql.exec.vector.util.FakeDataReader;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.optimizer.physical.Vectorizer;
+import org.apache.hadoop.hive.ql.plan.ExprNodeColumnDesc;
+import org.apache.hadoop.hive.ql.plan.FilterDesc;
+import org.apache.hadoop.hive.ql.plan.OperatorDesc;
+import org.apache.hadoop.hive.ql.plan.TopNKeyDesc;
+import org.apache.hadoop.hive.ql.plan.VectorFilterDesc;
+import org.apache.hadoop.hive.ql.plan.VectorTopNKeyDesc;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Tests that are necessary until VectorizedRowBatch selected issue is being fixed:
+ * It has three fields to manage selected elements:
+ * - selected (array)
+ * - selectedInUse
+ * - size (selected size)
+ * <p>
+ * The tricky thing is those fields are all public: so it is possible to modify some of them without keeping the others
+ * in sync.
+ * The issue that is fixed and checked with these tests is when somebody modifies the selected array but forgets to
+ * set the size parameter: Because of that, the Operators can try to process with wrong size value and it can cause
+ * ArrayIndexOutOfBoundsException.
+ * </p>
+ * <p>
+ * Related ticket: HIVE-26992
+ * Those tests can be removed when those fields are public they can be modified with public methods that ensures to
+ * use them properly.
+ * </p>
+ * Ticket to fix the issue: HIVE-26993
+ */
+public class TestVectorOperationProcess {
+
+  HiveConf hiveConf = new HiveConf();
+
+  @Test
+  public void testVectorFilterHasSelectedSmallerThanBatch() throws HiveException {
+    ExprNodeColumnDesc col1Expr = new ExprNodeColumnDesc(Long.class, "col1", "table", false);
+    List<String> columns = new ArrayList<>();
+    columns.add("col1");
+    FilterDesc filterDesc = new FilterDesc();
+    filterDesc.setPredicate(col1Expr);
+    VectorFilterDesc vectorDesc = new VectorFilterDesc();
+
+    Operator<? extends OperatorDesc> filterOp =
+            OperatorFactory.get(new CompilationOpContext(), filterDesc);
+
+    VectorizationContext vc = new VectorizationContext("name", columns);
+
+    VectorFilterOperator vfo = (VectorFilterOperator) Vectorizer.vectorizeFilterOperator(filterOp, vc, vectorDesc);
+
+    vfo.initialize(hiveConf, null);
+    VectorExpression ve1 = new FilterLongColGreaterLongColumn(0,1);
+    VectorExpression ve2 = new FilterLongColEqualDoubleScalar(2, 0);
+    VectorExpression ve3 = new FilterExprAndExpr();
+    ve3.setChildExpressions(new VectorExpression[] {ve1, ve2});
+    vfo.setFilterCondition(ve3);
+
+    FakeDataReader fdr = new FakeDataReader(1024, 3);
+    VectorizedRowBatch vrg = fdr.getNext();
+
+    vrg.selected = new int[] { 1, 2, 3, 4};
+
+    vfo.process(vrg, 0);
+  }
+
+  @Test
+  public void testTopNHasSelectedSmallerThanBatch() throws HiveException {
```

Comment:
There should be a test for a TopNKey as well. Test name is not declarative.

Reply:
Do you have any suggestions?

Reply:
move it to the TestTopNKeyFilter test

Reply:
`TestTopNKeyFilter` tests `TopNKeyFilter` only which is wrapped by `TopNKey` operator. It provides the logic of caching the top n values and TNK decides whether to forward a row based on `TopNKeyFilter.canForward` method.

I think it is worth having these tests in a class like `TestVectorTopNKeyOperator` if the goal is to test whether selected array is filled properly.

Reply:
👍

Reply:
Introduced `TestVectorTopNKeyOperator`. 

I couldn't test if the selected array is filled properly as our issue is about the case when it is not filled properly.